domains/service.go

```diff
@@ -351,15 +352,75 @@ func (svc *service) RemoveEntityMembers(ctx context.Context, session authn.Sessi
 	return svc.ProvisionManageService.RemoveEntityMembers(ctx, session, entityID, members)
 }
 
-func (svc *service) RoleRemoveMembers(ctx context.Context, session authn.Session, entityID, roleID string, members []string) (err error) {
-	for _, member := range members {
-		if err := svc.repo.DeleteInvitation(ctx, member, entityID); err != nil && err != repoerr.ErrNotFound {
-			return err
+func (svc *service) RoleRemoveMembers(ctx context.Context, session authn.Session, entityID, roleID string, members []string) error {
+	if len(members) == 0 {
+		return svcerr.ErrMalformedEntity
+	}
+
+	ro, err := svc.repo.RetrieveEntityRole(ctx, entityID, roleID)
+	if err != nil {
+		return errors.Wrap(svcerr.ErrViewEntity, fmt.Errorf("failed to retrieve role '%s' for entity '%s': %w", roleID, entityID, err))
+	}
+
+	isBuiltInRole := false
+	if _, ok := svc.ProvisionManageService.BuiltInRoles[roles.BuiltInRoleName(ro.Name)]; ok {
+		isBuiltInRole = true
+	}
+
+	if isBuiltInRole {
+		membersPage, listErr := svc.repo.RoleListMembers(ctx, ro.ID, 0, 0)
+		if listErr != nil {
+			return errors.Wrap(svcerr.ErrViewEntity, fmt.Errorf("failed to list members for built-in role '%s' to check count: %w", ro.Name, listErr))
+		}
+
+		if membersPage.Total <= uint64(len(members)) {
+			return errors.Wrap(svcerr.ErrRemoveEntity, fmt.Errorf("built-in role '%s' must retain at least one member. Attempting to remove %d member(s) from %d would leave too few.", ro.Name, len(members), membersPage.Total))
 		}
 	}
+
+	for _, memberID := range members {
+		if err := svc.repo.DeleteInvitation(ctx, memberID, entityID); err != nil && err != repoerr.ErrNotFound {
```

Comment:
We can have repo call where we can delete All the Invitation at once something like 

```go 
svc.repo.DeleteUsersInvitation(ctx,entityID, memberIDs...)
```

This requires new function invitations repo like ```DeleteUsersInvitation(ctx,entityID, memberIDs...)```

Reply:
Create ticket for this https://github.com/absmach/supermq/issues/2924
We can solve this later in different PR